text/0001-agile-coretime.md

```diff
@@ -0,0 +1,486 @@
+
+The specific interface is properly described in RFC-5.
+
+### Detail
+
+#### Parameters
+
+This proposal includes a number of parameters which need not necessarily be fixed. They are stated here along with a value, and are either *suggested* or *specified*. If *suggested*, it is non-binding and the proposal should not be judged on the value since the governance mechanism of Polkadot is expected to specify/maintain it. If *specified*, then the proposal should be judged on the merit of the value as-is.
+
+| Name                | Value                        | |
+| ------------------- | ---------------------------- | ---------- |
+| `BULK_PERIOD`       | `28 * DAYS`                  | specified  |
+| `LEADIN_PERIOD`     | `14 * DAYS`                  | specified  |
+| `TIMESLICE`         | `8 * MINUTES`                | specified  |
+| `BULK_TARGET`       | `30`                         | suggested  |
+| `BULK_LIMIT`        | `45`                         | suggested  |
+| `RENEWAL_PRICE_CAP` | `Perbill::from_percent(2)`   | suggested  |
+
+#### Reservations
+
+The Broker-chain includes some governance-set reservations of Coretime; these cover every System-chain as well as the pre-existing leased chains.
+
+#### Regions
+
+A *Region* is an assignable period of Coretime with a known regularity.
+
+All Regions are associated with a unique *Core Index*, to identify which core the Region controls the assignment of.
+
+All Regions are also associated with a *Core Parts*, an 80-bit bitmap, to denote the regularity on which it may be scheduled on the core. If all bits are set in the Core Parts value, it is said to be *Complete*.
+
+All Regions have a span. Region spans are quantized into periods of `TIMESLICE` blocks; `BULK_PERIOD` divides into `TIMESLICE` a whole number of times.
+
+The `Timeslice` type is a `u32` which can be multiplied by `TIMESLICE` to give a `BlockNumber` value representing the same quantity in terms of (Relay-chain) blocks.
+
+Regions can be tasked to a ParaId or placed in the Instantaneous Coretime Pool. If they have yet to be placed or tasked, then they are fresh. Fresh Regions have an *Owner*.
+
+#### Bulk Sales
+
+A sale of Bulk Coretime occurs on the Broker Chain every `BULK_PERIOD` blocks.
+
+In every sale, a `BULK_LIMIT` of individual *Regions* are offered for sale at a particular Sale Price.
+
+The Regions offered for sale have the same span: they last exactly `BULK_PERIOD` blocks, and begin `LEADIN_PERIOD` blocks into the future at the time of the sale.
+
+The Regions offered for sale also have complete
+
+Each Region offered for sale has a different Core Index, ensuring that they each represent an independently allocatable resource on the Polkadot UC.
+
+After every sale, the Next Sale Price is set according to the Previous Sale Price and the number of Regions sold compared to the desired and maximum amount of Regions to be sold. See Price Setting for additional detail.
```

Comment:
This model works under the assumption of DOT price volatility within a certain range. If we see a month where DOT price surges drastically, we encounter a condition that can make the price of core for the upcoming month unaffordable for a number of parachains (e.g. because they acquire the DOT on demand and suddenly see themselves unable to pay the rent for the coming month).

In such a situation, we would see that BULK_TARGET is missed by a large margin. This would result in a price decrease for the coming month, but would still present a situation where parachains miss the opportunity to acquire bulk coretime in that month.

We could either account for this by making bulk coretime dynamic pricing more complex (e.g. reverse auction during the lead-in period), or ensure that in such a situation instantaneous coretime sales is robust enough to offer parachains a viable alternative for this month.

Not a trivial decision, but I think it will present parachain teams with an operational overkill/burnout potential, if they have to pull off the operations of securing instantaneous coretime for a month.

Additionally, as far as I see it we currently project that bulk coretime goes live before instantaneous coretime; so this might kill off parachains for a month if they miss their purchase.

Reply:
> because they acquire the DOT just in time and suddenly see themselves unable to pay the rent for the coming month

Then obviously the chains should be holding DOT to mitigate exchange risk.